### PR TITLE
update Azure/go-ansiterm to v0.0.0-20210617225240-d185dfc1b5a1

### LINKED ISSUE #102988: [Failing Test][sig-release] ci-kubernetes-build
#### Which jobs are failing:

ci-kubernetes-build 

#### Which test(s) are failing

* Overall (build-master) 

#### Since when has it been failing:

2021-06-15 06:07:32 +0000 UTC

#### Testgrid link:

https://testgrid.k8s.io/sig-release-master-blocking#build-master

#### Reason for failure:

```
[0618 09:09:35] windows/386: build started
# k8s.io/kubernetes/vendor/github.com/Azure/go-ansiterm/winterm
vendor/github.com/Azure/go-ansiterm/winterm/ansi.go:167:2: constant 4294967286 overflows int
vendor/github.com/Azure/go-ansiterm/winterm/ansi.go:169:2: constant 4294967285 overflows int
vendor/github.com/Azure/go-ansiterm/winterm/ansi.go:171:2: constant 4294967284 overflows int
!!! [0618 09:13:40] Call tree:
!!! [0618 09:13:40]  1: /go/src/k8s.io/kubernetes/hack/lib/golang.sh:719 kube::golang::build_some_binaries(...)
!!! [0618 09:13:40]  2: /go/src/k8s.io/kubernetes/hack/lib/golang.sh:855 kube::golang::build_binaries_for_platform(...)
!!! [0618 09:13:40]  3: hack/make-rules/build.sh:27 kube::golang::build_binaries(...)
!!! [0618 09:15:23] Call tree:
!!! [0618 09:15:23]  1: hack/make-rules/build.sh:27 kube::golang::build_binaries(...)
make[1]: *** [Makefile:92: all] Error 1
make: *** [Makefile:512: cross] Error 1
!!! [0618 09:15:25] Call tree:
!!! [0618 09:15:25]  1: build/../build/common.sh:472 kube::build::run_build_command_ex(...)
!!! [0618 09:15:25]  2: build/release.sh:36 kube::build::run_build_command(...)
make: *** [Makefile:409: release] Error 1
level=fatal msg="Failed to run:: running make release: command /usr/bin/make release did not succeed: !!! [0618 09:15:23] Call tree:\n!!! [0618 09:15:23]  1: hack/make-rules/build.sh:27 kube::golang::build_binaries(...)\nmake[1]: *** [Makefile:92: all] Error 1\nmake: *** [Makefile:512: cross] Error 1\n!!! [0618 09:15:25] Call tree:\n!!! [0618 09:15:25]  1: build/../build/common.sh:472 kube::build::run_build_command_ex(...)\n!!! [0618 09:15:25]  2: build/release.sh:36 kube::build::run_build_command(...)\nmake: *** [Makefile:409: release] Error 1\n"
wrapper.sh] [TEST] Test Command exit code: 1
```

#### Anything else we need to know:

/sig release

### PR DESCRIPTION
**Pull request by BenTheElder**
fixes constant overflow compiling for windows/386

#### What type of PR is this?

/kind bug

#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes/kubernetes/issues/102988 <-- we don't have passing release builds right now (!)

#### Special notes for your reviewer:

see: https://github.com/Azure/go-ansiterm/commit/d185dfc1b5a126116ea5a19e148e29d16b4574c9

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```
/sig windows
/sig release
/priority critical-urgent

### CODE REVIEW

File: staging/src/k8s.io/api/go.mod [hunk 1]
```diff
@@ -11,6 +11,7 @@ require (
 )
 
 replace (
+	golang.org/x/sys => golang.org/x/sys v0.0.0-20210603081109-ebe580a85c40
```
Comment (BenTheElder) [inline staging/src/k8s.io/api/go.mod:14]: not sure why we got all of these suddenly. I ran `hack/pin-dependency.sh` for the azure dep, then `hack/update-vendor.sh`, added changes to git. nothing else. go 1.16.5 darwin/amd64 from homebrew.

Comment (BenTheElder) [reply staging/src/k8s.io/api/go.mod:14]: must be https://github.com/kubernetes/kubernetes/pull/103010/files#diff-1b28dbbe1a53e9df49d0a1ea840546007c9b3d84d3bc36a3b2f43c8c8ce15a37R5

**Comment by BenTheElder** [comment]
it's too bad hack/update-vendor.sh isn't smart enough to do https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/103010/pull-kubernetes-dependencies/1406012178892328960 for me automatically:
```
These modules are pinned to versions different than the minimal preferred version.
That means that without replace directives, a different version would be selected,
which breaks consumers of our published modules.
1. Use hack/pin-dependency.sh to switch to the preferred version for each module
2. Run hack/update-vendor.sh to rebuild the vendor directory
3. Run hack/lint-dependencies.sh to verify no additional changes are required

golang.org/x/sys
    pinned:    v0.0.0-20210603081109-ebe580a85c40
    preferred: v0.0.0-20210616094352-59db8d763f22
    hack/pin-dependency.sh golang.org/x/sys v0.0.0-20210616094352-59db8d763f22
All pinned versions of checked dependencies match their preferred version.
```

File: staging/src/k8s.io/api/go.mod (hunk 1 above)
Comment (BenTheElder) [reply staging/src/k8s.io/api/go.mod]: answer: https://github.com/kubernetes/kubernetes/pull/103010#issuecomment-864300216

**Comment by aojea** [comment]
```
golang.org/x/sys
    pinned:    v0.0.0-20210603081109-ebe580a85c40
    preferred: v0.0.0-20210616094352-59db8d763f22
    hack/pin-dependency.sh golang.org/x/sys v0.0.0-20210616094352-59db8d763f22
All pinned versions of checked dependencies match their preferred version.
```

**Comment by BenTheElder** [comment]
ran steps from above https://github.com/kubernetes/kubernetes/pull/103010#issuecomment-864300216. `hack/lint-dependencies.sh` reports nothing now.

/test pull-kubernetes-cross

**Comment by BenTheElder** [comment]
deps job is happy, but the real test here is https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/103010/pull-kubernetes-cross/1406015495873761280

**Comment by BenTheElder** [comment]
I guess lint-deps and update-vendor don't check internal modules ..?

added a `./hack/update-internal-modules.sh` to the commit

**Comment by dims** [comment]
/assign @liggitt

/approve 
(not that i'd make a difference as i don't have enough karma!)

**Comment by BenTheElder** [comment]
```

1/41 Tests Failed. | expand_less
-- | --
verify: staticcheck&nbsp;expand_less6m18sErrors from staticcheck: cluster/images/etcd/migrate/versions.go:129:1: this linter directive didn't match anything; should it be removed? () cluster/images/etcd/migrate/versions.go:192:1: this linter directive didn't match anything; should it be removed? ()  Please review the above warnings. You can test via:   hack/verify-staticcheck.sh &lt;failing package&gt; If the above warnings do not make sense, you can exempt the line or file. See:   https://staticcheck.io/docs/#ignoring-problems | verify: staticcheck&nbsp;expand_less | 6m18s | Errors from staticcheck: cluster/images/etcd/migrate/versions.go:129:1: this linter directive didn't match anything; should it be removed? () cluster/images/etcd/migrate/versions.go:192:1: this linter directive didn't match anything; should it be removed? ()  Please review the above warnings. You can test via:   hack/verify-staticcheck.sh &lt;failing package&gt; If the above warnings do not make sense, you can exempt the line or file. See:   https://staticcheck.io/docs/#ignoring-problems
verify: staticcheck&nbsp;expand_less | 6m18s
Errors from staticcheck: cluster/images/etcd/migrate/versions.go:129:1: this linter directive didn't match anything; should it be removed? () cluster/images/etcd/migrate/versions.go:192:1: this linter directive didn't match anything; should it be removed? ()  Please review the above warnings. You can test via:   hack/verify-staticcheck.sh &lt;failing package&gt; If the above warnings do not make sense, you can exempt the line or file. See:   https://staticcheck.io/docs/#ignoring-problems


```
🤨 
/retest

**Comment by BenTheElder** [comment]
cross build is green 👍 

verify is a known linter flake class with "is this method used or dead code" + methods used by unit tests

**Comment by BenTheElder** [comment]
still needs a /lgtm for merge. tests are good now.

**Pull request merged**